pkg/util/asyncworker/async_workers.go

```diff
@@ -116,6 +116,17 @@ func (aws *AsyncWorkers) handleWork(ctx context.Context, workName string, work *
 	defer func() {
 		if r := recover(); r != nil {
 			handleErr = fmt.Errorf("recover from %v", r)
+
+			metricErr := EmitCustomizedAsyncedMetrics(ctx,
```

Comment:
will workName always be a valid metric-name? will it contain some non-supported characters?

Reply:
In addition to existing EmitAsyncedMetrics, I add another function EmitCustomizedAsyncedMetrics in this pr. And it accept metric name as parameter. I use specified metric "name async_work_duration_ms" and "async_work_panic" in this pr and they are both valid names.